pkg/csi/cinder/controllerserver.go

```diff
@@ -194,12 +198,18 @@ func (cs *controllerServer) ControllerUnpublishVolume(ctx context.Context, req *
 	err := cs.Cloud.DetachVolume(instanceID, volumeID)
 	if err != nil {
 		klog.V(3).Infof("Failed to DetachVolume: %v", err)
+		if cpoerrors.IsNotFound(err) {
```

Comment:
I think here you are checking for NotFound and then creating a new NotFound error. Is this just to change the error message?  (same for the other diff a few lines below)

Reply:
@anguslees Its because according to csi spec , it should return standard gRPC status , which may be not be the case if we return directly , so the same has been updated here.